r/tests/testthat/test-convert-array.R

```diff
@@ -893,6 +907,22 @@ test_that("convert to vector works for blob::blob()", {
   )
 })
 
+test_that("convert to vector works for binary_view -> blob::blob()", {
+  skip_if_not_installed("blob")
+
+  array <- as_nanoarrow_array(letters, schema = na_binary_view())
+  expect_identical(
+    convert_array(array, blob::blob()),
+    blob::as_blob(lapply(letters, charToRaw))
+  )
+
+  array_with_nulls <- as_nanoarrow_array(c(letters, NA), schema = na_binary_view())
```

Comment:
Should this also work? It doesn't for me:

```r
# Test all nulls
> array_with_nulls <- as_nanoarrow_array(c(rep(NA, 10)), schema = na_binary_view())
Error: NotImplemented: binary_view
```

Reply:
ah, so this is similar to other calls, like,

```r
> as_nanoarrow_array(-1, schema = na_binary_view())
Error: NotImplemented: binary_view
```

which is an error bubbled up from the arrow package:

```
> traceback()
10: vec_to_Array(x, type)
9: as_arrow_array.default(x, type = arrow::as_data_type(schema))
8: arrow::as_arrow_array(x, type = arrow::as_data_type(schema)) at array.R#43
7: as_nanoarrow_array(arrow::as_arrow_array(x, type = arrow::as_data_type(schema))) at as-array.R#40
6: as_nanoarrow_array.default(x, schema = schema, .from_c = TRUE) at array.R#43
5: as_nanoarrow_array(x, schema = schema, .from_c = TRUE) at as-array.R#428
4: as_nanoarrow_array_from_c(c(NA, NA, NA, NA, NA, NA, NA, NA, NA,
   NA), <pointer: 0x600002e68280>)
3: .Call(nanoarrow_c_as_array_default, x, schema) at as-array.R#60
2: as_nanoarrow_array.default(rep(NA, 10), schema = na_string_view()) at array.R#43
1: as_nanoarrow_array(rep(NA, 10), schema = na_string_view())
>
```

So maybe this isn't something to change?

Reply:
sounds good